pytorch_lightning/trainer/distrib_data_parallel.py

```diff
@@ -373,6 +373,14 @@ def ddp_train(self, process_idx, model):
             model, optimizers = model.configure_apex(amp, model, self.optimizers, self.amp_level)
             self.optimizers = optimizers
 
+            # Reinitialize optimizer.step properties added by schedulers
+            for lr_scheduler in self.lr_schedulers:
+                for optimizer in self.optimizers:
+                    scheduler = lr_scheduler['scheduler']
+                    # check that we dont mix users optimizers and schedulers
+                    if scheduler.optimizer == optimizer:
+                        scheduler.__class__.__mro__[-2].__init__(scheduler, optimizer)
```

Comment:
Are you sure, this is always -2? If I have my own fancy scheduler, that also derives from some other class, we could maybe have a different mro?

Reply:
Did not think about that, this kind of assumes that everything was just derived *only* from `torch.optim.lr_schedulers._LRScheduler`. You are correct, since the mro of this stupid scheduler:
```
class MyScheduler(torch.optim.lr_scheduler.LambdaLR, torch.utils.data.Dataset):
    pass
```
have mro:
```
(__main__.MyScheduler,
 torch.optim.lr_scheduler.LambdaLR,
 torch.optim.lr_scheduler._LRScheduler,
 torch.utils.data.dataset.Dataset,
 object)
```
where the correct index then would be -3 (or 2). Have an idea how to fix this?